lightningd/lightningd.c

```diff
@@ -574,17 +574,17 @@ void notify_new_block(struct lightningd *ld, u32 block_height)
 	channel_notify_new_block(ld, block_height);
 }
 
-static void on_sigint(int _ UNUSED)
+static void on_sigint(int _)
 {
         static const char *msg = "lightningd: SIGINT caught, exiting.\n";
-        write(STDERR_FILENO, msg, strlen(msg));
+        _ = write(STDERR_FILENO, msg, strlen(msg));
```

Comment:
I prefer to fix this using "write_all" which specifically doesn't have this problem.

Or at least put in a snarky reference to glibc+gcc braindamage a-la https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66425 :)